{"2":"1","3":"Ju9C-1621846617594","4":{"version":1,"incompatibleVersion":0,"fv":"0"},"5":[{"3":"3060-1621846615933","5":[{"2":"2","3":"p5PQ-1621846617594","7":[{"8":"异常的两种处理方式：try catch         "},{"8":" throws：处理异常的方式，使用在方法声明的末尾","9":[{"0":"#F33232","2":"c"}]}]}]},{"3":"VvDs-1656393750379","4":{"version":1,"u":"https://note.youdao.com/yws/res/21263/WEBRESOURCE41fc079f520d0f7fedf1ddc8a2d92346","w":641,"h":848},"6":"im"},{"3":"P0oM-1651734456958","4":{"l":"h3"},"5":[{"2":"2","3":"bel7-1651734456955","7":[{"8":"异常体系结构"}]}],"6":"h"},{"3":"vvTb-1656393801423","5":[{"2":"2","3":"zlaE-1656393801422","7":[{"8":"1.","9":[{"2":"b"},{"0":"#000000","2":"c"},{"0":"Arial","2":"ff"}]},{"8":"运行时异常","9":[{"2":"b"},{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"zo1v-1656393832045","5":[{"2":"2","3":"wVtS-1656393832044","7":[{"8":" ","9":[{"0":"#000000","2":"c"}]},{"8":"是指编译器不要求强制处置的异常。一般是指编程时的逻辑错误，是程序","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"XPXl-1656393832048","5":[{"2":"2","3":"KoA8-1656393832047","7":[{"8":"员应该积极避免其出现的异常。","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]},{"8":"java.lang.RuntimeException","9":[{"2":"b"},{"0":"#c00000","2":"c"},{"0":"Arial","2":"ff"}]},{"8":"类及它的子","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"WYoF-1656393832050","5":[{"2":"2","3":"7rpG-1656393832049","7":[{"8":"类都是运行时异常。","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"eakL-1656393832052","5":[{"2":"2","3":"oDXh-1656393832051","7":[{"8":" ","9":[{"0":"#000000","2":"c"}]},{"8":"对于这类异常，可以不作处理，因为这类异常很普遍，若全处理可能会对","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"uXvA-1656393832054","5":[{"2":"2","3":"WJXJ-1656393832053","7":[{"8":"程序的可读性和运行效率产生影响。","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"TGAU-1656393832056","5":[{"2":"2","3":"WxQb-1656393832055","7":[{"8":"2.","9":[{"2":"b"},{"0":"#000000","2":"c"},{"0":"Arial","2":"ff"}]},{"8":"编译时异常","9":[{"2":"b"},{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"6SY8-1656393832058","5":[{"2":"2","3":"lVxD-1656393832057","7":[{"8":" ","9":[{"0":"#000000","2":"c"}]},{"8":"是指编译器要求必须处置的异常。即程序在运行时由于外界因素造成的一","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"wZqh-1656393832060","5":[{"2":"2","3":"qlKf-1656393832059","7":[{"8":"般性异常。","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]},{"8":"编译器要求","9":[{"0":"#c00000","2":"c"},{"0":"SimSun","2":"ff"}]},{"8":"Java","9":[{"0":"#c00000","2":"c"},{"0":"Arial","2":"ff"}]},{"8":"程序必须捕获或声明所有编译时异常。","9":[{"0":"#c00000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"R5jt-1656393832062","5":[{"2":"2","3":"NyTJ-1656393832061","7":[{"8":" ","9":[{"0":"#000000","2":"c"}]},{"8":"对于这类异常，如果程序不处理，可能会带来意想不到的结果","9":[{"0":"#000000","2":"c"},{"0":"SimSun","2":"ff"}]}]}]},{"3":"19kn-1656393801779","4":{"l":"h3"},"5":[{"2":"2","3":"Kx9Q-1656393801777","7":[{"8":"手动抛出异常："}]}],"6":"h"},{"3":"3Z9W-1651734528576","5":[{"2":"2","3":"vDxA-1651734528569","7":[{"8":"throw：","9":[{"0":"#F33232","2":"c"}]},{"8":"手动生成一个异常对象，并抛出，使用在方法中。"}]}]},{"3":"PUlM-1651734587102","4":{"l":"h3"},"5":[{"2":"2","3":"sW2M-1651734587101","7":[{"8":"捕获异常："}]}],"6":"h"},{"3":"lqdE-1651734457506","5":[{"2":"2","3":"ZSuS-1651734457503","7":[{"8":"try{} catch(){}"}]}]},{"3":"pDzW-1651735237383","5":[{"2":"2","3":"eiD4-1651735237380","7":[{"8":"catch(){} finally{}"}]}]},{"3":"GrQQ-1651735209074","5":[{"2":"2","3":"UU41-1651735209068","7":[{"8":"finally是可选的。使用try将可能出现异常代码包装起来，在执行过程中，一旦出现异常，就会生成一个对应异常类的对象，根据此对象的类型，去catch中进行匹配。一旦try中的异常对象匹配到某一个catch时，就进入catch中进行异常的处理。一旦处理完成，就跳出当前的try-catch结构（在没有写finally的情况)"}]}]},{"3":"Tc3P-1651735209577","5":[{"2":"2","3":"bXpg-1651735209574","7":[{"8":" 在try结构中声明的变量，再出了try结构以后，就不能再被调用"}]}]},{"3":"axC7-1651736070795","5":[{"2":"2","3":"U9tC-1651736070781","7":[{"8":"trycatch 结构可以嵌套"}]}]},{"3":"Lekr-1651736070957","5":[{"2":"2","3":"EdAB-1651736070954","7":[{"8":"使用try-catch-finally处理编译时异常，是得程序在编译时就不再报错，但是运行时仍可能报锋相当于我们使用try-catch-finally将一个编译时可能出现的异常，延迟到运行时出现。"}]}]},{"3":"co5E-1651736071131","5":[{"2":"2","3":"ea3I-1651736071129","7":[{"8":"开发中，由于运行时异常比较常见，所以我们通常就不针对运行时异常编写try-catch-finally"}]}]},{"3":"55p8-1651736307612","5":[{"2":"2","3":"fZrB-1651736307610","7":[{"8":"针对于编译时异常，我们说一定要考虑异常的处理。"}]}]},{"3":"aapn-1651738222106","5":[{"2":"2","3":"G62g-1651738222103"}]},{"3":"YPM7-1651738222284","5":[{"2":"2","3":"fpQ1-1651738222282","7":[{"8":"rhrows 异常类型"}]}]},{"3":"osAJ-1651736789317","4":{"version":1},"5":[{"2":"2","3":"okrz-1651736789313","7":[{"8":"程序在正常执行的过程中，一旦出现异常，就会在异常代码处生成一个对应异常类的对象并将此对象抛出。一旦抛出异常，其后的代码便不再执行"}]}]},{"3":"DaEu-1651738472961","4":{"version":1},"5":[{"2":"2","3":"UPkp-1651738472959"}]},{"3":"gMOH-1651738242550","4":{"l":"h3"},"5":[{"2":"2","3":"vy5z-1651738242546","7":[{"8":"用户自定义异常："}]}],"6":"h"},{"3":"9oTx-1651738884344","4":{"wrap":false,"la":"java","th":"tomorrow"},"5":[{"3":"YMjS-1651738884342","5":[{"2":"2","3":"WP9z-1651738470904","7":[{"8":"public class MyException extends RuntimeException{"}]}],"6":"cl"},{"3":"fU8g-1651738885208","5":[{"2":"2","3":"6Ajd-1651738885207","7":[{"8":"    static final long serialVersionUID=-293748923759258234L;"}]}],"6":"cl"},{"3":"y1WP-1651738889843","5":[{"2":"2","3":"6HZc-1651738889841","7":[{"8":"}"}]}],"6":"cl"}],"6":"cd"},{"3":"Fqdn-1651738884345","5":[{"2":"2","3":"AvCL-1651738884346","7":[{"8":"1.继承现有的异常结构2.提供全局常量serialVersionUID3.提供构造方法"}]}]},{"3":"jfJ3-1651738099594","4":{"version":1},"6":"hr"},{"3":"PZBX-1651738082114","5":[{"2":"2","3":"the3-1651738082110"}]},{"3":"cFlC-1651738082301","5":[{"2":"2","3":"bMnv-1651738082299","7":[{"8":"如果父类中被重写的方法没有throws方式处理异常，则子类重写的方法也不能使用throws，意味着如果子类重写的方法中有异常，必须使用try-catch-finally方式处理。、"}]}]},{"3":"Yy1J-1656393589490","5":[{"2":"2","3":"tZYN-1656393589488"}]},{"3":"4s38-1656393589649","4":{"l":"h3"},"5":[{"2":"2","3":"95F4-1656393589647","7":[{"8":"try catch加强：自动资源关闭操作"}]}],"6":"h"},{"3":"LavC-1656388595186","4":{"version":1,"u":"https://note.youdao.com/yws/res/21264/WEBRESOURCE7359983505899476af2364e6fe83858f","w":723,"h":270},"6":"im"},{"3":"n12K-1656388867073","4":{"version":1,"u":"https://note.youdao.com/yws/res/21265/WEBRESOURCE6ca69dd539d34eea873b53a8c5853783","w":715,"h":310},"6":"im"},{"3":"HU5o-1656388745121","4":{"version":1},"5":[{"2":"2","3":"H3Dy-1656388745120","7":[{"8":"Java8中，可以实现资源的自动关闭，但是要求执行后必须关闭的所有资源必须在try子句中初始化，否则编译不通过。"}]}]}],"title":"","__compress__":true}